document_generator/full_text_document_generator.py

```diff
@@ -70,24 +71,23 @@ def create_allfields_field(fullrecord_field: str) -> dict:
     @staticmethod
     def txt_files_2_full_text(zip_doc: zipfile.ZipFile):
         """
-        Read all the .TXT files in the input folder and concatenate the content of all of them in a single string
-        :return: String with the content of all the .TXT files
+        Read all .TXT files in a zip and concatenate their contents.
+        :return: Single string with all text files concatenated.
         """
-        # TODO: Check MACOSX files in the logs to know if the ZIP files with this folder is being processed correctly
-        # and if the full text is being generated correctly
-        # check the logs te see the among of files with this messages to determine if we need the line
-        # if not i_file.startswith('__MACOSX/') or if we need to include a resource forks in the code
-        # python library to handle resource forks => phttps://pypi.org/project/rsrcfork/
-        full_text_parts = []
-        macosx_files = False
-        for i_file in zip_doc.namelist():
-            if not i_file.startswith('__MACOSX/'):
-                if zip_doc.getinfo(i_file).filename.endswith(".txt"):
-                    full_text_parts.append(string_preparation(zip_doc.read(i_file)))
-            else:
-                macosx_files = True
+        macosx_files = any(i_file.startswith('__MACOSX/') for i_file in zip_doc.namelist())
+
+        # Get only .txt files and sort them by name
+        txt_files = sorted([i_file for i_file in zip_doc.namelist() if
+                     i_file.endswith('.txt') and not i_file.startswith('__MACOSX/')])
+
+        # Process files in parallel
+        with ThreadPoolExecutor() as executor:
```

Comment:
If I understand correctly, this is reading each text file out of the zip in parallel, and then concatenating the results?